var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -207,6 +207,10 @@ def setup_dependent_package(self, module, dep_spec):
         self.spec.mpicxx = join_path(self.prefix.bin, 'mpicxx')
         self.spec.mpifc  = join_path(self.prefix.bin, 'mpif90')
         self.spec.mpif77 = join_path(self.prefix.bin, 'mpif77')
+        if '+slurm' in self.spec:
+            self.spec.run = join_path(self.prefix.bin, 'srun')
```

Comment:
`srun` will be part of slurm, so not in the prefix. I guess we can work something with an external `slurm` package and `spec['slurm'].prefix.bin`
For cray I think there is a `arun` wrapper